ServerCore/Pages/Hints/CreateBulkMulti.cshtml.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Hints
+{
+    [Authorize(Policy = "IsEventAdminOrEventAuthor")]
+    public class CreateBulkMultiModel : EventSpecificPageModel
+    {
+        public CreateBulkMultiModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        [BindProperty]
+        public bool DeleteExisting { get; set; }
+
+        [BindProperty]
+        public string PuzzleName { get; set; }
+
+        [BindProperty]
+        public string Description { get; set; }
+
+        [BindProperty]
+        public string HintContent { get; set; }
+
+        [BindProperty]
+        public string Cost { get; set; }
+
+        [BindProperty]
+        public string DisplayOrder { get; set; }
+
+        public IActionResult OnGet()
+        {
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            StringReader puzzleNameReader = new StringReader(PuzzleName);
+            StringReader descriptionReader = new StringReader(Description ?? string.Empty);
+            StringReader contentReader = new StringReader(HintContent ?? string.Empty);
+            StringReader costReader = new StringReader(Cost ?? string.Empty);
+            StringReader displayOrderReader = new StringReader(DisplayOrder ?? string.Empty);
+
+            Dictionary<string, int> puzzleTitleLookup = new Dictionary<string, int>();
+
+            while (true)
+            {
+                string puzzleName = puzzleNameReader.ReadLine();
+                string description = descriptionReader.ReadLine();
+                string content = contentReader.ReadLine();
+                string cost = costReader.ReadLine();
+                string displayOrder = displayOrderReader.ReadLine();
+
+                int puzzleId = -1;
+
+                if (puzzleName != null && !puzzleTitleLookup.TryGetValue(puzzleName, out puzzleId))
+                {
+                    Puzzle puzzle = await (from Puzzle p in _context.Puzzles
+                                           where p.Name == puzzleName && p.EventID == this.Event.ID
```

Comment:
I'm not sure if there's much we can do, but this will be case and white-space sensitive

Reply:
Agree, though it'll generate errors in those cases that the author can respond to.